ams/server/src/main/java/com/netease/arctic/server/table/TableRuntime.java

```diff
@@ -256,6 +265,21 @@ public void completeProcess(boolean success) {
   }
 
   private void updateOptimizingStatus(OptimizingStatus status) {
+    SelfOptimizingStatusDurationMsContent selfOptimizingStatusDurationMsContent =
+        new SelfOptimizingStatusDurationMsContent(
+            tableIdentifier.toString(), optimizingStatus.displayValue());
+    selfOptimizingStatusDurationMsContent
+        .tableOptimizingStatusDurationMs()
+        .inc(System.currentTimeMillis() - currentStatusStartTime);
+    if (optimizingStatus == OptimizingStatus.COMMITTING) {
+      selfOptimizingStatusDurationMsContent.setOptimizingType(
+          optimizingProcess.getOptimizingType().name());
+      selfOptimizingStatusDurationMsContent.setOptimizingProcessId(
+          optimizingProcess.getProcessId());
+      selfOptimizingStatusDurationMsContent.setTargetSnapshotId(
+          optimizingProcess.getTargetSnapshotId());
+    }
+    metricsManager.emit(selfOptimizingStatusDurationMsContent);
```

Comment:
Do we should calculate the relevant metrics for the optimizing process in `completeProcess()`?

Reply:
Yes, this indicator is missing here

Reply:
I think I should filter out the logic from `optimizing` to `COMMITTING` when the state changes. The `optimizing` cost is calculated separately in the SelfOptimizingTotalCostMsContent indicator.
WDYT @hameizi

Reply:
I think that the duration of the previous state can be calculated for every state transition, including the transition from "optimizing" to "COMMITTING". The total duration of "optimizing" should include the duration of the "COMMITTING" state, so I previously suggested calculating the total duration of the optimizing process in the completeProcess() method.

Reply:
solved.